File: semester-1/quicky/src/honk.java

public class honk {

	private static boolean f1(boolean a, boolean b, boolean c, boolean d) {
		return (a && b && c) || (a && !b && c && d) || (a && !b && c && !d) || (a && !b && !c && !d) || (!a && b && c && d) || (!a && b && c && !d) || (!a && !b && !c && !d) || (!a && !b && c) || (a && b && !c && d) || (a && b && !c && !d);
	}

	private static boolean f2(boolean a, boolean b, boolean c, boolean d) {
		return (!b && !d) || c || (a && b);
	}

	public static void main(String args[]) {

		for(int m = 0; m < 2; m ++) {

			boolean a;
			if (m == 0) a = true;
			else a = false;

			for(int n = 0; n < 2; n ++) {

				boolean b;
				if (n == 0) b = true;
				else b = false;

				for(int x = 0; x < 2; x ++) {

					boolean c;
					if (x == 0) c = true;
					else c = false;

					for(int y = 0; y < 2; y ++) {

						boolean d;
						if (y == 0) d = true;
						else d = false;


						System.out.println("f1 a:"+m+" b:"+n+" c:"+x+" d:"+y+"   "+ f1(a,b,c,d));
						System.out.println("f2 a:"+m+" b:"+n+" c:"+x+" d:"+y+"   "+ f2(a,b,c,d));


						System.out.println();

					}
				}
			}

		}
	}
}

File: semester-1/18.10.29-helloworld/honk.rs
fn main() {

    // Here in rust you don't have to explicitely dereference, which is nice.
    // & is the reference operator, it creates a borrow.
    // &mut creates the mutable borrow.
    // It's all quite nice indeed.


    struct Honk {
        msg: i32,
    }

    let pete = Honk { msg: 22 };
    let refpete = &pete;
    println!("{}", refpete.msg);

    let elle = 24;
    let rofi = &elle;

    println!("{}", elle);
    println!("{}", rofi);
    println!("{}", &rofi);

    println!("honk");
}
